File: sdk/python-sdk/test/protocols/test_UpdateEndpoint.py
import pytest

from test.test_utils import get_test_config, cleanup
from verity_sdk.protocols.v0_6.UpdateEndpoint import UpdateEndpoint
from verity_sdk.utils import EVERNYM_MSG_QUALIFIER
from verity_sdk.utils.Context import Context


@pytest.mark.asyncio
async def test_update_endpoint():
    context = await Context.create_with_config(await get_test_config())
    update_endpoint = UpdateEndpoint()
    msg = update_endpoint.update_endpoint_msg(context)

    assert msg['@type'] == '{}/configs/0.6/UPDATE_COM_METHOD'.format(EVERNYM_MSG_QUALIFIER)
    assert msg['@id'] is not None
    assert msg['comMethod']['id'] == 'webhook'
    assert msg['comMethod']['type'] == 2
    assert msg['comMethod']['value'] == context.endpoint_url
    assert msg['comMethod']['packaging']['pkgType'] == '1.0'
    assert msg['comMethod']['packaging']['recipientKeys'] == [context.sdk_verkey]

    await cleanup(context)


@pytest.mark.asyncio
async def test_update_endpoint_with_oauth2_v1_authentication():
    context = await Context.create_with_config(await get_test_config())
    update_endpoint = UpdateEndpoint(
        {
            'type': 'OAuth2',
            'version': 'v1',
            'data': {
                'url': 'https://auth.url/token',
                'grant_type': 'client_credentials',
                'client_id': 'ajeyKDizsDkwYeEmRmHU78gf7W3VIEoA',
                'client_secret': 'aaGxxcGi6kb6AxIe'
            }
        }
    )
    msg = update_endpoint.update_endpoint_msg(context)

    assert msg['@type'] == '{}/configs/0.6/UPDATE_COM_METHOD'.format(EVERNYM_MSG_QUALIFIER)
    assert msg['@id'] is not None
    assert msg['comMethod']['id'] == 'webhook'
    assert msg['comMethod']['type'] == 2
    assert msg['comMethod']['value'] == context.endpoint_url
    assert msg['comMethod']['packaging']['pkgType'] == '1.0'
    assert msg['comMethod']['packaging']['recipientKeys'] == [context.sdk_verkey]

    # authentication
    assert msg['comMethod']['authentication']['type'] == 'OAuth2'
    assert msg['comMethod']['authentication']['version'] == 'v1'
    assert msg['comMethod']['authentication']['data']['url'] == 'https://auth.url/token'
    assert msg['comMethod']['authentication']['data']['grant_type'] == 'client_credentials'
    assert msg['comMethod']['authentication']['data']['client_id'] == 'ajeyKDizsDkwYeEmRmHU78gf7W3VIEoA'
    assert msg['comMethod']['authentication']['data']['client_secret'] == 'aaGxxcGi6kb6AxIe'

    await cleanup(context)

File: sdk/nodejs-sdk/test/test_UpdateEndpoint.js
/* eslint-env mocha */
'use strict'
const chai = require('chai')
const expect = chai.expect
chai.use(require('chai-as-promised'))
// const Context = require('../src/utils/Context')
const UpdateEndpoint = require('../src/protocols/v0_6/UpdateEndpoint')
const helpers = require('../test/test_utils')

const authentication = {
  type: 'OAuth2',
  version: 'v1',
  data: {
    url: 'https://auth.url/token',
    grant_type: 'client_credentials',
    client_id: 'ajeyKDizsDkwYeEmRmHU78gf7W3VIEoA',
    client_secret: 'aaGxxcGi6kb6AxIe'
  }
}

describe('UpdateEndpoint', () => {
  it('should init correctly', async () => {
    const ue = new UpdateEndpoint(
      authentication
    )
    expect(ue.authentication).to.equal(authentication)
  })

  it('should build UPDATE_COM_METHOD msg correctly', async () => {
    const context = await helpers.getTestContext()
    const ue = new UpdateEndpoint()
    const msg = await ue.updateMsg(context)
    expect(msg['@id']).to.not.be.an('undefined')
    expect(msg['@type']).to.equal(
     `${ue.msgQualifier}/${ue.msgFamily}/${ue.msgFamilyVersion}/${ue.msgNames.UPDATE_ENDPOINT}`
    )
    expect(msg.comMethod.value).to.equal(context.endpointUrl)
    expect(msg.comMethod.id).to.equal('webhook')
    expect(msg.comMethod.type).to.equal(2)
    expect(msg.comMethod.packaging.pkgType).to.equal('1.0')
    expect(msg.comMethod.packaging.recipientKeys).to.eql([context.sdkVerKey])
    expect(msg.comMethod.authentication).to.an('undefined')
  }).timeout(5000)

  it('should build UPDATE_COM_METHOD msg with OAuth2 v1 authentication correctly', async () => {
    const context = await helpers.getTestContext()
    const ue = new UpdateEndpoint(
      {
        type: 'OAuth2',
        version: 'v1',
        data: {
          url: 'https://auth.url/token',
          grant_type: 'client_credentials',
          client_id: 'ajeyKDizsDkwYeEmRmHU78gf7W3VIEoA',
          client_secret: 'aaGxxcGi6kb6AxIe'
        }
      }
    )
    const msg = await ue.updateMsg(context)
    expect(msg['@id']).to.not.be.an('undefined')
    expect(msg['@type']).to.equal(
     `${ue.msgQualifier}/${ue.msgFamily}/${ue.msgFamilyVersion}/${ue.msgNames.UPDATE_ENDPOINT}`
    )
    expect(msg.comMethod.value).to.equal(context.endpointUrl)
    expect(msg.comMethod.id).to.equal('webhook')
    expect(msg.comMethod.type).to.equal(2)
    expect(msg.comMethod.packaging.pkgType).to.equal('1.0')
    expect(msg.comMethod.packaging.recipientKeys).to.eql([context.sdkVerKey])

    // authentication
    expect(msg.comMethod.authentication.type).to.equal('OAuth2')
    expect(msg.comMethod.authentication.version).to.equal('v1')
    expect(msg.comMethod.authentication.data.url).to.equal('https://auth.url/token')
    expect(msg.comMethod.authentication.data.grant_type).to.equal('client_credentials')
    expect(msg.comMethod.authentication.data.client_id).to.equal('ajeyKDizsDkwYeEmRmHU78gf7W3VIEoA')
    expect(msg.comMethod.authentication.data.client_secret).to.equal('aaGxxcGi6kb6AxIe')
  }).timeout(5000)
})
